File: 1С/ПР/ИНФ 1К1С/8/1.py
from Stack import *
    
S = Stack()
a = "привет"
half = len(a)//2

for char in a[half:]:
    S.push(char)
for char in a[:half]:
    S.push(char)

while not S.isEmpty():
    print(S.pop(),end="")

File: 1С/ПР/ТИП 1К1С/14ПР/1.cpp
#include "stack.h"
#include <iostream>
#include <string>
using namespace std;

bool isCorrect(string &data, Stack<char> st)
{
    int unclosed = 0;
    char val{};
    while (!st.isEmpty())
    {
        val = st.pop();
        if (val == ')')
            unclosed++;
        else
            unclosed--;

        if (unclosed < 0)
            return false;
    }
    return unclosed == 0;
}

int main(int argc, char const *argv[])
{
    string data = "(())()";
    Stack st = Stack<char>();
    for (char ch : data)
    {
        st.push(ch);
    }
    cout << data << endl;
    cout << isCorrect(data, st) << endl;
    return 0;
}
